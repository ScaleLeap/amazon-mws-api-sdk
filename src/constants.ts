/**
 * The version value is automatically replaced during builds.
 */
export const USER_AGENT = '@scaleleap/amazon-mws-api-sdk/1.9.0'
