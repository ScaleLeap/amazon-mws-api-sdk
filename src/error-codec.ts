import { Codec, exactly, oneOf, optional, string } from 'purify-ts/Codec'

export enum CodeEnum {
  InputStreamDisconnected,
  InvalidParameterValue,
  AccessDenied,
  InvalidAccessKeyId,
  SignatureDoesNotMatch,
  InvalidAddress,
  InternalError,
  'Internal Error',
  QuotaExceeded,
  RequestThrottled,
  ResourceNotFound,
  ScheduledPackageAlreadyExists,
  RegionNotSupported,
  ScheduleWindowExpired,
  InvalidOrderState,
  PickupSlotNotAvailable,
  AccessToFeedProcessingResultDenied,
  ContentMD5Missing,
  ContentMD5DoesNotMatch,
  FeedCanceled,
  FeedProcessingResultNoLongerAvailable,
  FeedProcessingResultNotReady,
  InputDataError,
  InvalidFeedSubmissionId,
  InvalidFeedType,
  InvalidRequest,
  NonRetriableInternalError,
  RetriableInternalError,
  AccessToReportDenied,
  InvalidReportId,
  InvalidReportRequestId,
  InvalidReportType,
  InvalidScheduleFrequency,
  ReportNoLongerAvailable,
  ReportNotReady,
  DependencyFatalException,
  DependencyRetriableException,
  DependencyUnauthorizedException,
  InternalErrorFatalException,
  InvalidInputFatalException,
}

export const Error = Codec.interface({
  Type: string,
  Code: oneOf(Object.keys(CodeEnum).map((x) => exactly(x)) as [Codec<string>, ...Codec<string>[]]),
  Message: string,
  Detail: optional(string),
})

export const MWSApiError = Codec.interface({
  ErrorResponse: Codec.interface({
    Error,
    RequestID: optional(string),
    RequestId: optional(string),
  }),
})
