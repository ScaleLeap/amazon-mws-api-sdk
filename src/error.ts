/* eslint-disable max-classes-per-file */
export class MWSError extends Error {
  constructor(...parameters: string[]) {
    // Propagate some vendor-specific arguments
    super(...parameters)

    // If someone downlevels the compilation target to es5
    Object.setPrototypeOf(this, MWSError.prototype)

    // Maintains proper stack trace (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, MWSError)
    }
  }
}

export class HttpError extends MWSError {
  public message = 'Encountered an error while sending a request'

  constructor(public error: unknown, ...parameters: string[]) {
    super(...parameters)
    Object.setPrototypeOf(this, HttpError.prototype)

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, HttpError)
    }
  }
}

export class ParsingError extends MWSError {
  public message = 'Encountered an error while parsing a response'

  constructor(public error: string, ...parameters: string[]) {
    super(...parameters)
    Object.setPrototypeOf(this, ParsingError.prototype)

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, ParsingError)
    }
  }
}
/* eslint-enable max-classes-per-file */
