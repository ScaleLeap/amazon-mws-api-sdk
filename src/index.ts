export * from './http'
export * from './mws'
export * from './sections/sellers'
export * from './sections/orders'
export * from './sections/products/products'
export * from './sections/fulfillment-inventory'
export * from './sections/reports'
export * from './error'
export * from '@scaleleap/amazon-marketplaces'
