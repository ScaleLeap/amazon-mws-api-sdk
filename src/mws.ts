import { HttpClient } from './http'
import { FulfillmentInventory } from './sections/fulfillment-inventory'
import { Orders } from './sections/orders'
import { Products } from './sections/products/products'
import { Reports } from './sections/reports'
import { Sellers } from './sections/sellers'

export class MWS {
  private _sellers!: Sellers

  private _orders!: Orders

  private _fulfillmentInventory!: FulfillmentInventory

  private _products!: Products

  private _reports!: Reports

  constructor(private httpClient: HttpClient) {}

  get sellers() {
    if (!this._sellers) {
      this._sellers = new Sellers(this.httpClient)
    }

    return this._sellers
  }

  get orders() {
    if (!this._orders) {
      this._orders = new Orders(this.httpClient)
    }

    return this._orders
  }

  get fulfillmentInventory() {
    if (!this._fulfillmentInventory) {
      this._fulfillmentInventory = new FulfillmentInventory(this.httpClient)
    }

    return this._fulfillmentInventory
  }

  get products() {
    if (!this._products) {
      this._products = new Products(this.httpClient)
    }

    return this._products
  }

  get reports() {
    if (!this._reports) {
      this._reports = new Reports(this.httpClient)
    }

    return this._reports
  }
}
