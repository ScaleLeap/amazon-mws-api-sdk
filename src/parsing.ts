/** A collection of parsing utils */

export const ensureArray = <T>(x: T | T[]): T[] => (Array.isArray(x) ? x : [x])

export const parseBoolean = (x: 'Yes' | 'No'): boolean => {
  switch (x) {
    case 'Yes':
      return true
    case 'No':
      return false
    default:
      throw new Error('TODO, but should be library-specific error')
  }
}
