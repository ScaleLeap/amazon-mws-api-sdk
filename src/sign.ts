import crypto from 'crypto'

export const sign = (queryString: string, secret: string): string => {
  const hmac = crypto.createHmac('sha256', secret)
  hmac.update(queryString, 'utf8')
  return hmac.digest('base64')
}
