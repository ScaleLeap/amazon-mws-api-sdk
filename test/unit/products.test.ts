import { amazonMarketplaces, HttpClient, ParsingError } from '../../src'
import { MWS } from '../../src/mws'
// import { NextToken } from '../../src/parsing'
import { getFixture } from '../utils'

const httpConfig = {
  awsAccessKeyId: '',
  marketplace: amazonMarketplaces.CA,
  mwsAuthToken: '',
  secretKey: '',
  sellerId: '',
}

const headers = {
  'x-mws-request-id': '0',
  'x-mws-timestamp': '2020-05-06T09:22:23.582Z',
  'x-mws-quota-max': '1000',
  'x-mws-quota-remaining': '999',
  'x-mws-quota-resetson': '2020-04-06T10:22:23.582Z',
}

const mockListMatchingProducts = new MWS(
  new HttpClient(httpConfig, () =>
    Promise.resolve({
      data: getFixture('products_list_matching_products'),
      headers,
    }),
  ),
)

const mockGetMyFeesEstimate = new MWS(
  new HttpClient(httpConfig, () =>
    Promise.resolve({
      data: getFixture('products_get_my_fees_estimate'),
      headers,
    }),
  ),
)

const mockMwsFail = new MWS(
  new HttpClient(httpConfig, () => Promise.resolve({ data: '', headers: {} })),
)

// const mockNextTokenInventorySupply = new NextToken('ListInventorySupply', '123')

const parsingError = 'Expected an object, but received a string with value ""'

describe('products', () => {
  describe('listMatchingProducts', () => {
    it('returns an array of products when the response is valid', async () => {
      expect.assertions(1)

      expect(
        await mockListMatchingProducts.products.listMatchingProducts({
          MarketplaceId: '',
          Query: '',
        }),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the response is not valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.products.listMatchingProducts({ MarketplaceId: '', Query: '' }),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getMyFeesEstimate', () => {
    it('returns a list fee estimates when the response is valid', async () => {
      expect.assertions(1)

      expect(
        await mockGetMyFeesEstimate.products.getMyFeesEstimate({ FeesEstimateRequestList: [] }),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the response is not valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.products.getMyFeesEstimate({ FeesEstimateRequestList: [] }),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })
})
