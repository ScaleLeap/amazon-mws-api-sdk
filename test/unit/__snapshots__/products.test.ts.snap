// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products getCompetitivePricingForASIN returns product and competitive prices when response is valid 1`] = `
Array [
  Array [
    Object {
      "Product": Object {
        "CompetitivePricing": Object {
          "CompetitivePrices": Object {
            "CompetitivePrice": Object {
              "CompetitivePriceId": 1,
              "Price": Object {
                "LandedPrice": Object {
                  "Amount": 34.27,
                  "CurrencyCode": "USD",
                },
                "ListingPrice": Object {
                  "Amount": 34.27,
                  "CurrencyCode": "USD",
                },
                "Shipping": Object {
                  "Amount": 0,
                  "CurrencyCode": "USD",
                },
              },
            },
          },
          "NumberOfOfferListings": Object {
            "OfferListingCount": Array [
              11,
              1,
              10,
            ],
          },
        },
        "Identifiers": Object {
          "MarketplaceASIN": Object {
            "ASIN": "B002KT3XQM",
            "MarketplaceId": "ATVPDKIKX0DER",
          },
        },
        "SalesRankings": "",
      },
    },
  ],
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": "2020-04-06T10:22:23.582Z",
    "requestId": "0",
    "timestamp": "2020-05-06T09:22:23.582Z",
  },
]
`;

exports[`products getCompetitivePricingForSku returns product and competitive prices when response is valid 1`] = `
Array [
  Array [
    Object {
      "Product": Object {
        "CompetitivePricing": Object {
          "CompetitivePrices": Object {
            "CompetitivePrice": Array [
              Object {
                "CompetitivePriceId": 1,
                "Price": Object {
                  "LandedPrice": Object {
                    "Amount": 40.03,
                    "CurrencyCode": "USD",
                  },
                  "ListingPrice": Object {
                    "Amount": 40.03,
                    "CurrencyCode": "USD",
                  },
                  "Shipping": Object {
                    "Amount": 0,
                    "CurrencyCode": "USD",
                  },
                },
              },
              Object {
                "CompetitivePriceId": 2,
                "Price": Object {
                  "LandedPrice": Object {
                    "Amount": 30.5,
                    "CurrencyCode": "USD",
                  },
                  "ListingPrice": Object {
                    "Amount": 30.5,
                    "CurrencyCode": "USD",
                  },
                  "Shipping": Object {
                    "Amount": 0,
                    "CurrencyCode": "USD",
                  },
                },
              },
            ],
          },
          "NumberOfOfferListings": Object {
            "OfferListingCount": Array [
              296,
              145,
              151,
            ],
          },
          "TradeInValue": Object {
            "Amount": 17.05,
            "CurrencyCode": "USD",
          },
        },
        "Identifiers": Object {
          "MarketplaceASIN": Object {
            "ASIN": 1933890517,
            "MarketplaceId": "ATVPDKIKX0DER",
          },
          "SKUIdentifier": Object {
            "MarketplaceId": "ATVPDKIKX0DER",
            "SellerId": "A1IMEXAMPLEWRC",
            "SellerSKU": "SKU2468",
          },
        },
        "SalesRankings": Object {
          "SalesRank": Array [
            Object {
              "ProductCategoryId": "book_display_on_website",
              "Rank": 900,
            },
            Object {
              "ProductCategoryId": 271578011,
              "Rank": 1,
            },
            Object {
              "ProductCategoryId": 355562011,
              "Rank": 8,
            },
            Object {
              "ProductCategoryId": 173516,
              "Rank": 25,
            },
          ],
        },
      },
    },
  ],
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": "2020-04-06T10:22:23.582Z",
    "requestId": "0",
    "timestamp": "2020-05-06T09:22:23.582Z",
  },
]
`;

exports[`products getMatchingProduct returns a matching product when the response is valid 1`] = `
Array [
  Array [
    Object {
      "Product": Object {
        "AttributeSets": Object {
          "ns2:ItemAttributes": Object {
            "ns2:Binding": "Apparel",
            "ns2:Brand": "Pearl iZUMi",
            "ns2:Department": "mens",
            "ns2:Feature": Array [
              "Select transfer fabric sets the benchmark for moisture transfer and
                        four-way performance stretch",
              "6-Panel anatomic design for superior, chafe-free comfort, UPF 50+ sun
                        protection",
              "Comfortable silicone leg grippers keep shorts in place,
                        moisture-wicking, antimicrobial properties",
              "Tour 3D Chamois is male-specific and non-chafing with padding in
                        key areas",
              "86 percent nylon, 14% spandex, 9-Inch inseam",
            ],
            "ns2:ItemDimensions": Object {
              "ns2:Height": 2,
              "ns2:Length": 9,
              "ns2:Width": 9,
            },
            "ns2:Label": "Pearl iZUMi",
            "ns2:ListPrice": Object {
              "ns2:Amount": 50,
              "ns2:CurrencyCode": "USD",
            },
            "ns2:Manufacturer": "Pearl iZUMi",
            "ns2:Model": 275,
            "ns2:PackageDimensions": Object {
              "ns2:Height": 2.8,
              "ns2:Length": 9.75,
              "ns2:Weight": 0.4,
              "ns2:Width": 8.4,
            },
            "ns2:PackageQuantity": 1,
            "ns2:PartNumber": 275,
            "ns2:ProductGroup": "Apparel",
            "ns2:ProductTypeName": "SHORTS",
            "ns2:Publisher": "Pearl iZUMi",
            "ns2:SmallImage": Object {
              "ns2:Height": 75,
              "ns2:URL": "http://ecx.images-amazon.com/images/I/41ty3Sn%2BU8L._SL75_.jpg",
              "ns2:Width": 58,
            },
            "ns2:Studio": "Pearl iZUMi",
            "ns2:Title": "Pearl iZUMi Men's Quest Cycling Short",
          },
        },
        "Identifiers": Object {
          "MarketplaceASIN": Object {
            "ASIN": "B002KT3XRQ",
            "MarketplaceId": "ATVPDKIKX0DER",
          },
        },
        "Relationships": Object {
          "ns2:VariationChild": Array [
            Object {
              "Identifiers": Object {
                "MarketplaceASIN": Object {
                  "ASIN": "B002KT3XQC",
                  "MarketplaceId": "ATVPDKIKX0DER",
                },
              },
              "ns2:Color": "Black",
              "ns2:Size": "Small",
            },
            Object {
              "Identifiers": Object {
                "MarketplaceASIN": Object {
                  "ASIN": "B002KT3XQW",
                  "MarketplaceId": "ATVPDKIKX0DER",
                },
              },
              "ns2:Color": "Black",
              "ns2:Size": "Medium",
            },
            Object {
              "Identifiers": Object {
                "MarketplaceASIN": Object {
                  "ASIN": "B002KT3XQM",
                  "MarketplaceId": "ATVPDKIKX0DER",
                },
              },
              "ns2:Color": "Black",
              "ns2:Size": "Large",
            },
            Object {
              "Identifiers": Object {
                "MarketplaceASIN": Object {
                  "ASIN": "B002KT3XR6",
                  "MarketplaceId": "ATVPDKIKX0DER",
                },
              },
              "ns2:Color": "Black",
              "ns2:Size": "X-Large",
            },
            Object {
              "Identifiers": Object {
                "MarketplaceASIN": Object {
                  "ASIN": "B002KT3XRG",
                  "MarketplaceId": "ATVPDKIKX0DER",
                },
              },
              "ns2:Color": "Black",
              "ns2:Size": "XX-Large",
            },
          ],
        },
        "SalesRankings": Object {
          "SalesRank": Array [
            Object {
              "ProductCategoryId": "apparel_display_on_website",
              "Rank": 159,
            },
            Object {
              "ProductCategoryId": 2420095011,
              "Rank": 1,
            },
            Object {
              "ProductCategoryId": 2611189011,
              "Rank": 4,
            },
          ],
        },
      },
    },
  ],
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": "2020-04-06T10:22:23.582Z",
    "requestId": "0",
    "timestamp": "2020-05-06T09:22:23.582Z",
  },
]
`;

exports[`products getMatchingProductForId returns a matching product when the response is valid 1`] = `
Array [
  Array [
    Object {
      "Products": Array [
        Object {
          "AttributeSets": Object {
            "ns2:ItemAttributes": Object {
              "ns2:Author": Array [
                "Marmanis, Haralambos",
                "Babenko, Dmitry",
              ],
              "ns2:Binding": "Paperback",
              "ns2:Edition": 1,
              "ns2:IsEligibleForTradeIn": true,
              "ns2:ItemDimensions": Object {
                "ns2:Height": 9.17,
                "ns2:Length": 7.36,
                "ns2:Weight": 1.4,
                "ns2:Width": 0.75,
              },
              "ns2:Label": "Manning Publications",
              "ns2:Languages": Object {
                "ns2:Language": Array [
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Unknown",
                  },
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Original Language",
                  },
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Published",
                  },
                ],
              },
              "ns2:ListPrice": Object {
                "ns2:Amount": 44.99,
                "ns2:CurrencyCode": "USD",
              },
              "ns2:Manufacturer": "Manning Publications",
              "ns2:NumberOfItems": 1,
              "ns2:NumberOfPages": 368,
              "ns2:PackageDimensions": Object {
                "ns2:Height": 0.8,
                "ns2:Length": 9.1,
                "ns2:Weight": 1.35,
                "ns2:Width": 7.3,
              },
              "ns2:ProductGroup": "Book",
              "ns2:ProductTypeName": "ABIS_BOOK",
              "ns2:PublicationDate": "2009-07-05",
              "ns2:Publisher": "Manning Publications",
              "ns2:SmallImage": Object {
                "ns2:Height": 75,
                "ns2:URL": "http://ecx.images-amazon.com/images/I/51EEz05N2HL._SL75_.jpg",
                "ns2:Width": 60,
              },
              "ns2:Studio": "Manning Publications",
              "ns2:Title": "Algorithms of the Intelligent Web",
            },
          },
          "Identifiers": Object {
            "MarketplaceASIN": Object {
              "ASIN": 1933988665,
              "MarketplaceId": "ATVPDKIKX0DER",
            },
          },
          "Relationships": "",
          "SalesRankings": Object {
            "SalesRank": Array [
              Object {
                "ProductCategoryId": "book_display_on_website",
                "Rank": 59485,
              },
              Object {
                "ProductCategoryId": 377886011,
                "Rank": 32,
              },
              Object {
                "ProductCategoryId": 3887,
                "Rank": 66,
              },
              Object {
                "ProductCategoryId": 3870,
                "Rank": 82,
              },
            ],
          },
        },
      ],
    },
    Object {
      "Products": Array [
        Object {
          "AttributeSets": Object {
            "ns2:ItemAttributes": Object {
              "ns2:Author": "Rowling, J.K.",
              "ns2:Binding": "Paperback",
              "ns2:Brand": "Scholastic Press",
              "ns2:Creator": "GrandPrÃ©, Mary",
              "ns2:Edition": "1st",
              "ns2:Feature": "Recommended Age: 9 years and up",
              "ns2:IsAutographed": false,
              "ns2:IsMemorabilia": false,
              "ns2:ItemDimensions": Object {
                "ns2:Height": 0.8,
                "ns2:Length": 7.5,
                "ns2:Weight": 0.5,
                "ns2:Width": 5.2,
              },
              "ns2:Label": "Scholastic Paperbacks",
              "ns2:Languages": Object {
                "ns2:Language": Array [
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Unknown",
                  },
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Original Language",
                  },
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Published",
                  },
                ],
              },
              "ns2:ListPrice": Object {
                "ns2:Amount": 10.99,
                "ns2:CurrencyCode": "USD",
              },
              "ns2:Manufacturer": "Scholastic Paperbacks",
              "ns2:NumberOfItems": 1,
              "ns2:NumberOfPages": 320,
              "ns2:PackageDimensions": Object {
                "ns2:Height": 1,
                "ns2:Length": 7.5,
                "ns2:Weight": 0.5,
                "ns2:Width": 5.2,
              },
              "ns2:PackageQuantity": 1,
              "ns2:PartNumber": 9780590353427,
              "ns2:ProductGroup": "Book",
              "ns2:ProductTypeName": "ABIS_BOOK",
              "ns2:PublicationDate": "1999-10-01",
              "ns2:Publisher": "Scholastic Paperbacks",
              "ns2:ReleaseDate": "1999-09-08",
              "ns2:SmallImage": Object {
                "ns2:Height": 75,
                "ns2:URL": "http://ecx.images-amazon.com/images/I/51MU5VilKpL._SL75_.jpg",
                "ns2:Width": 51,
              },
              "ns2:Studio": "Scholastic Paperbacks",
              "ns2:Title": "Harry Potter and the Sorcerer's Stone (Book 1)",
            },
          },
          "Identifiers": Object {
            "MarketplaceASIN": Object {
              "ASIN": "059035342X",
              "MarketplaceId": "ATVPDKIKX0DER",
            },
          },
          "Relationships": "",
          "SalesRankings": Object {
            "SalesRank": Array [
              Object {
                "ProductCategoryId": "book_display_on_website",
                "Rank": 362,
              },
              Object {
                "ProductCategoryId": 15356791,
                "Rank": 6,
              },
              Object {
                "ProductCategoryId": 3153,
                "Rank": 9,
              },
              Object {
                "ProductCategoryId": 3045,
                "Rank": 14,
              },
            ],
          },
        },
        Object {
          "AttributeSets": Object {
            "ns2:ItemAttributes": Object {
              "ns2:Author": "J.K. Rowling, Mary GrandPrÃƒÂ©",
              "ns2:Binding": "Paperback",
              "ns2:Label": "Scholastic Paperbacks",
              "ns2:Languages": Object {
                "ns2:Language": Array [
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Unknown",
                  },
                  Object {
                    "ns2:Name": "english",
                    "ns2:Type": "Published",
                  },
                ],
              },
              "ns2:Manufacturer": "Scholastic Paperbacks",
              "ns2:NumberOfItems": 1,
              "ns2:NumberOfPages": 784,
              "ns2:PackageDimensions": Object {
                "ns2:Height": 0.73,
                "ns2:Length": 8.43,
                "ns2:Weight": 0.79,
                "ns2:Width": 5.85,
              },
              "ns2:ProductGroup": "Book",
              "ns2:ProductTypeName": "BOOKS_1973_AND_LATER",
              "ns2:PublicationDate": "1999-09-08",
              "ns2:Publisher": "Scholastic Paperbacks",
              "ns2:SmallImage": Object {
                "ns2:Height": 75,
                "ns2:URL": "http://ecx.images-amazon.com/images/I/51STfI7UiCL._SL75_.jpg",
                "ns2:Width": 52,
              },
              "ns2:Studio": "Scholastic Paperbacks",
              "ns2:Title": "Harry Potter and the Sorcerer's Stone",
            },
          },
          "Identifiers": Object {
            "MarketplaceASIN": Object {
              "ASIN": 439708184,
              "MarketplaceId": "ATVPDKIKX0DER",
            },
          },
          "Relationships": "",
          "SalesRankings": Object {
            "SalesRank": Object {
              "ProductCategoryId": "book_display_on_website",
              "Rank": 2091024,
            },
          },
        },
      ],
    },
  ],
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": "2020-04-06T10:22:23.582Z",
    "requestId": "0",
    "timestamp": "2020-05-06T09:22:23.582Z",
  },
]
`;

exports[`products getMyFeesEstimate returns a list fee estimates when the response is valid 1`] = `
Array [
  Object {
    "FeesEstimateResultList": Array [
      Object {
        "Error": undefined,
        "FeesEstimate": Object {
          "FeeDetailList": Array [
            Object {
              "FeeAmount": Object {
                "Amount": 8.7,
                "CurrencyCode": "USD",
              },
              "FeePromotion": Object {
                "Amount": 1,
                "CurrencyCode": "USD",
              },
              "FeeType": "ReferralFee",
              "FinalFee": Object {
                "Amount": 7.7,
                "CurrencyCode": "USD",
              },
              "IncludedFeeDetailList": undefined,
              "TaxAmount": undefined,
            },
            Object {
              "FeeAmount": Object {
                "Amount": 0,
                "CurrencyCode": "USD",
              },
              "FeePromotion": undefined,
              "FeeType": "VariableClosingFee",
              "FinalFee": Object {
                "Amount": 0,
                "CurrencyCode": "USD",
              },
              "IncludedFeeDetailList": undefined,
              "TaxAmount": undefined,
            },
            Object {
              "FeeAmount": Object {
                "Amount": 2.3,
                "CurrencyCode": "USD",
              },
              "FeePromotion": undefined,
              "FeeType": "FulfillmentFees",
              "FinalFee": Object {
                "Amount": 2.3,
                "CurrencyCode": "USD",
              },
              "IncludedFeeDetailList": undefined,
              "TaxAmount": undefined,
            },
          ],
          "TimeOfFeesEstimation": 2015-07-19T23:15:11.859Z,
          "TotalFeesEstimate": Object {
            "Amount": 10,
            "CurrencyCode": "USD",
          },
        },
        "FeesEstimateIdentifier": Object {
          "IdType": "ASIN",
          "IdValue": "B0002GTTRC",
          "IsAmazonFulfilled": true,
          "MarketplaceId": "ATVPDKIKX0DER",
          "PriceToEstimateFees": Object {
            "ListingPrice": Object {
              "Amount": 58,
              "CurrencyCode": "USD",
            },
            "Points": Object {
              "PointsMonetaryValue": Object {
                "Amount": 100,
                "CurrencyCode": "USD",
              },
              "PointsNumber": 0,
            },
            "Shipping": Object {
              "Amount": 0,
              "CurrencyCode": "USD",
            },
          },
          "SellerInputIdentifier": "IDDDDDDDD",
        },
        "Status": "Success",
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": "2020-04-06T10:22:23.582Z",
    "requestId": "0",
    "timestamp": "2020-05-06T09:22:23.582Z",
  },
]
`;

exports[`products listMatchingProducts returns an array of products when the response is valid 1`] = `
Array [
  Object {
    "Products": Array [
      Object {
        "AttributeSets": Object {
          "ns2:ItemAttributes": Object {
            "ns2:Author": "Rowling, J.K.",
            "ns2:Binding": "Paperback",
            "ns2:Brand": "Scholastic Press",
            "ns2:Creator": "GrandPrÃ©, Mary",
            "ns2:Edition": "1st",
            "ns2:Feature": "Recommended Age: 9 years and up",
            "ns2:IsAutographed": false,
            "ns2:IsMemorabilia": false,
            "ns2:ItemDimensions": Object {
              "ns2:Height": 0.8,
              "ns2:Length": 7.5,
              "ns2:Weight": 0.5,
              "ns2:Width": 5.2,
            },
            "ns2:Label": "Scholastic Paperbacks",
            "ns2:Languages": Object {
              "ns2:Language": Array [
                Object {
                  "ns2:Name": "english",
                  "ns2:Type": "Unknown",
                },
                Object {
                  "ns2:Name": "english",
                  "ns2:Type": "Original Language",
                },
                Object {
                  "ns2:Name": "english",
                  "ns2:Type": "Published",
                },
              ],
            },
            "ns2:ListPrice": Object {
              "ns2:Amount": 10.99,
              "ns2:CurrencyCode": "USD",
            },
            "ns2:Manufacturer": "Scholastic Paperbacks",
            "ns2:NumberOfItems": 1,
            "ns2:NumberOfPages": 320,
            "ns2:PackageDimensions": Object {
              "ns2:Height": 1,
              "ns2:Length": 7.5,
              "ns2:Weight": 0.5,
              "ns2:Width": 5.2,
            },
            "ns2:PackageQuantity": 1,
            "ns2:PartNumber": 9780590353427,
            "ns2:ProductGroup": "Book",
            "ns2:ProductTypeName": "ABIS_BOOK",
            "ns2:PublicationDate": "1999-10-01",
            "ns2:Publisher": "Scholastic Paperbacks",
            "ns2:ReleaseDate": "1999-09-08",
            "ns2:SmallImage": Object {
              "ns2:Height": 75,
              "ns2:URL": "http://ecx.images-amazon.com/images/I/51MU5VilKpL._SL75_.jpg",
              "ns2:Width": 51,
            },
            "ns2:Studio": "Scholastic Paperbacks",
            "ns2:Title": "Harry Potter and the Sorcerer's Stone (Book 1)",
          },
        },
        "Identifiers": Object {
          "MarketplaceASIN": Object {
            "ASIN": "059035342X",
            "MarketplaceId": "ATVPDKIKX0DER",
          },
        },
        "Relationships": "",
        "SalesRankings": Object {
          "SalesRank": Array [
            Object {
              "ProductCategoryId": "book_display_on_website",
              "Rank": 401,
            },
            Object {
              "ProductCategoryId": 15356791,
              "Rank": 5,
            },
            Object {
              "ProductCategoryId": 3153,
              "Rank": 8,
            },
            Object {
              "ProductCategoryId": 17468,
              "Rank": 16,
            },
          ],
        },
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": "2020-04-06T10:22:23.582Z",
    "requestId": "0",
    "timestamp": "2020-05-06T09:22:23.582Z",
  },
]
`;
